authenticationsdk/util/jwt/JWTUtility.py

```diff
@@ -0,0 +1,304 @@
+        except json.JSONDecodeError as parse_err:
+            raise InvalidJwkException('Invalid public key JSON format', parse_err)
+    elif isinstance(public_key, dict) and 'kty' in public_key:
+        jwk_key = public_key
+    else:
+        raise InvalidJwkException(ERROR_MESSAGES['INVALID_PUBLIC_KEY_FORMAT'])
+    
+    if jwk_key.get('kty') != 'RSA':
+        raise InvalidJwkException(ERROR_MESSAGES['INVALID_RSA_KEY'])
+    
+    if not jwk_key.get('n') or not jwk_key.get('e'):
+        raise InvalidJwkException(ERROR_MESSAGES['MISSING_RSA_PARAMS'])
+    
+    return jwk_key
+
+
+def _convert_jwk_to_public_key(jwk_key):
+    """
+    Convert JWK RSA parameters to a cryptography RSA public key object.
+    
+    Args:
+        jwk_key (dict): The JWK object with RSA parameters
+        
+    Returns:
+        RSAPublicKey: The cryptography RSA public key object
+        
+    Raises:
+        InvalidJwkException: If key conversion fails
+    """
+    try:
+        # Decode base64url encoded n and e parameters
+        n_bytes = _base64url_decode(jwk_key['n'])
+        e_bytes = _base64url_decode(jwk_key['e'])
+    except Exception as decode_err:
+        raise InvalidJwkException('Invalid base64url encoding in JWK parameters', decode_err)
+    
+    try:
+        # Convert bytes to integers
+        n = int.from_bytes(n_bytes, byteorder='big')
+        e = int.from_bytes(e_bytes, byteorder='big')
+        
+        # Create RSA public key
+        public_numbers = rsa.RSAPublicNumbers(e, n)
+        public_key = public_numbers.public_key(default_backend())
+        
+        return public_key
+    except Exception as key_err:
+        raise InvalidJwkException('Failed to create RSA public key from JWK', key_err)
+
+
+def parse(jwt_token):
+    """
+    Parse a JWT token and extract its header, payload, and signature components.
+    
+    Args:
+        jwt_token (str): The JWT token to parse
+        
+    Returns:
+        dict: Dictionary containing:
+            - header (dict): Decoded JWT header
+            - payload (dict): Decoded JWT payload
+            - signature (str): Base64url encoded signature
+            - raw_header (str): Raw base64url encoded header
+            - raw_payload (str): Raw base64url encoded payload
+            
+    Raises:
+        InvalidJwtException: If the JWT token is invalid or malformed
+    """
+    if jwt_token is None:
+        raise InvalidJwtException('JWT token is null or undefined')
+    
+    if not isinstance(jwt_token, str):
+        raise InvalidJwtException('JWT token must be a string')
+    
+    token_parts = jwt_token.split('.')
+    if len(token_parts) != 3:
+        raise InvalidJwtException('Invalid JWT token format: expected 3 parts separated by dots')
+    
+    # Validate that all parts are non-empty
+    if not token_parts[0] or not token_parts[1] or not token_parts[2]:
+        raise InvalidJwtException('Invalid JWT token: one or more parts are empty')
```

Comment:
Change this error message. It should not say that certain parts are missing. If some part is missing, it is a malformed JWT.

Say "Malformed JWT : JWT provided does not conform to the proper structure for JWT".